frame/support/procedural/src/pallet/parse/pallet_struct.rs

```diff
@@ -39,12 +40,30 @@ pub struct PalletStructDef {
 	pub store: Option<(syn::Visibility, keyword::Store)>,
 	/// The span of the pallet::pallet attribute.
 	pub attr_span: proc_macro2::Span,
+	/// Whether to specify the storages max encoded len when implementing `StoragesInfo`.
+	/// Contains the span of the attribute.
+	pub generate_storages_info: Option<proc_macro2::Span>,
 }
 
-/// Parse for `#[pallet::generate_store($vis trait Store)]`
-pub struct PalletStructAttr {
-	vis: syn::Visibility,
-	keyword: keyword::Store,
+/// Parse for one variant of:
+/// * `#[pallet::generate_store($vis trait Store)]`
+/// * `#[pallet::generate_storages_info]`
```

Comment:
```suggestion
/// * `#[pallet::generate_storage_info]`
```

Why `storages`? Its a very strange word

Reply:
should I rename the trait StorageInfo too ?
The attribute means it implement the trait `StoragesInfo` because it is the info of all pallet storages. But yeah name is not very good maybe I should come up with better name, but I have 3 items: StorageInfo (type returned for each storage), StorageInfoTrait (trait implemented by each storage in order to return the StorageInfo), and StoragesInfo (trait implemented for each pallet in order to return the info for all storages)

Reply:
I think it should just be singular, unless it represents more than one thing.

Then it could be `AllStorageInfo` or similar... but i think the singular makes sense. Storage is already plural.

Reply:
Any of `AllStorageInfo`, `MemberStorageInfo`, `MultiStorageInfo`, or similar would be better in my opinion than `StoragesInfo`. I'd like to see that trait (and its method) renamed to something more grammatical.

Reply:
Indeed `AllStorageInfo` seems better, but finally I changed my mind and merged it with StorageInfoTrait.

The idea is that in the future some storage can need to return multiple info. Like if a storage store the number of item in the storage in another prefix then it needs to return 2 StorageInfo:
the information about the values after some prefix, and the information about this counter which is on another prefix.

Maybe it is even better to rename StorageInfo to StorageInfoPart then

Reply:
note that this makes implementation less performant as it needs to allocate and pass vec everywhere. But I think it is fine as this is only for benchmarking.